### PR TITLE
Redesign WnDropdownSelector with expanding animation

### PR DESCRIPTION
**Pull request by erskingardner**
## Summary
- Replace Flutter's DropdownButton with custom expanding dropdown component
- Add size variants (small: 44px, large: 56px) matching Figma design specs
- Implement 120ms expanding animation (container expands in-place instead of overlay)
- Add checkmark for selected/hovered items with proper icon alignment
- Add helper text, error state, and disabled state support
- Toggle chevron to X icon when dropdown is open

https://github.com/user-attachments/assets/9084f33e-aebd-489f-923e-e7978cdbb452

## Summary by CodeRabbit

* **New Features**
  * Redesigned dropdown selector with animated expand/collapse, header-driven interaction, selectable checkmarks, responsive item heights, size variants, helper text, error and disabled visuals.

* **Tests**
  * Expanded UI tests covering open/close flows, selection and checkmark behavior, icon/hover states, size/helper/error/disabled variations, scrolling, edge cases, and updated app settings tests to use the new selector.

* **Documentation**
  * Added guidance on comments and preferring hooks/providers over StatefulWidgets.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: lib/widgets/wn_dropdown_selector.dart [hunk 1]
```diff
@@ -13,80 +14,272 @@ class WnDropdownOption<T> {
...
+        isOpen.value = false;
+        animationController.reverse();
+      }
+      return null;
+    }, [isDisabled]);
+
+    final selectedLabel = useMemoized(() {
+      final selected = options.where((o) => o.value == value);
+      return selected.isNotEmpty ? selected.first.label : '';
+    }, [options, value]);
```
[the remark this answers is not in the document]
Comment (erskingardner) [reply lib/widgets/wn_dropdown_selector.dart]: To be honest, I find the where pattern that we have easier to understand. I think I'd prefer to leave this one alone.
